pkg/prometheus/collector.go

```diff
@@ -84,7 +85,6 @@ func (c *Collector) collectPrometheus(ch chan<- prometheus.Metric, p v1.Promethe
 	if cpf.EnforcedSampleLimit != nil {
 		ch <- prometheus.MustNewConstMetric(descPrometheusEnforcedSampleLimit, prometheus.GaugeValue, float64(*cpf.EnforcedSampleLimit), namespace, name)
 	}
-	if cpf.Shards != nil {
-		ch <- prometheus.MustNewConstMetric(descPrometheusSpecShards, prometheus.GaugeValue, float64(*cpf.Shards), namespace, name)
-	}
+
+	ch <- prometheus.MustNewConstMetric(descPrometheusSpecShards, prometheus.GaugeValue, float64(ptr.Deref(cpf.Shards, 1)), namespace, name)
```

Comment:
@simonpasquier interesting side note - the `prometheus_operator_spec_shards` metric was only present when the number of shards is set on the CRD - the metric should exist for all values I think.

Reply:
good catch!